mapper/orm/src/test/java/org/hibernate/search/mapper/orm/OrmExtensionTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.search.engine.backend.document.converter.runtime.FromIndexFieldValueConvertContext;
+import org.hibernate.search.engine.backend.document.converter.runtime.spi.FromIndexFieldValueConvertContextImpl;
+import org.hibernate.search.mapper.orm.session.context.impl.HibernateOrmSessionContextImpl;
+import org.hibernate.search.mapper.pojo.bridge.runtime.IdentifierBridgeFromDocumentIdentifierContext;
+import org.hibernate.search.mapper.pojo.bridge.runtime.PropertyBridgeWriteContext;
+import org.hibernate.search.mapper.pojo.bridge.runtime.RoutingKeyBridgeToRoutingKeyContext;
+import org.hibernate.search.mapper.pojo.bridge.runtime.TypeBridgeWriteContext;
+import org.hibernate.search.mapper.pojo.bridge.runtime.impl.BridgeSessionContextImpl;
+
+import org.junit.Test;
+
+import org.easymock.EasyMockSupport;
+
+public class OrmExtensionTest extends EasyMockSupport {
+	private final SessionImplementor sessionImplementor = createMock( SessionImplementor.class );
+	private final HibernateOrmSessionContextImpl sessionContext = new HibernateOrmSessionContextImpl( sessionImplementor );
+
+	@Test
+	public void identifierBridge() {
+		IdentifierBridgeFromDocumentIdentifierContext fromDocumentContext = new BridgeSessionContextImpl( sessionContext );
+		resetAll();
+		replayAll();
+		assertThat( fromDocumentContext.extension( OrmExtension.get() ) ).isSameAs( sessionContext );
```

Comment:
It means the user will have to do that to get the ORM session?

Same remark about the EasyMock machinery being harmful here.

Reply:
The user will have to call `fromDocumentContext.extension( OrmExtension.get() )`, yes.

About EasyMock, if you don't want to use it and want to implement `SessionImplementor` instead, be me guest, but I doubt you do ;)